packages/expo-calendar/src/Calendar.ts

```diff
@@ -558,6 +558,149 @@ export type DaysOfTheWeek = {
 
 export { PermissionResponse, PermissionStatus, PermissionHookOptions };
 
+/**
+ * The result of presenting the calendar dialog.
+ * These constants map to [`EKEventEditViewAction`](https://developer.apple.com/documentation/eventkitui/ekeventeditviewaction) and [`EKEventViewAction`](https://developer.apple.com/documentation/eventkitui/ekeventviewaction).
+ * @platform ios
+ * */
+export type CalendarDialogResultActionsIos = 'done' | 'canceled' | 'deleted' | 'responded';
+
+/**
+ * The result of launching the calendar Activity.
+ * Android doesn't provide enough information to determine the user's action, so the result is always 'done'.
+ * The user may have cancelled the dialog, saved the event, or deleted it.
+ * @platform android
+ * */
+export type CalendarDialogResultActionsAndroid = 'done';
+
+/**
+ * The result of presenting a calendar dialog. It indicates what action the user took (e.g. canceled the dialog).
+ * The information made available by Android is less complete compared to iOS.
+ * @header systemProvidedUI
+ * */
+export type CalendarDialogResult = {
+  action: CalendarDialogResultActionsIos | CalendarDialogResultActionsAndroid;
+};
+
+/**
+ * The result of creating an event in the calendar dialog on iOS.
+ * @platform ios
+ * */
+export type CalendarDialogCreatedEventResultIos = {
+  action: 'saved';
+  /**
+   * The ID of the newly created event.
+   * */
+  id: string;
```

Comment:
It could just be `null` on Android
```suggestion
  /**
   * The ID of the newly created event.
   */
  id: string | null;
```